src/utils/parsers/BaseParser.ts

```diff
@@ -31,31 +22,47 @@ export type MatchResult = {
     index: number
 }
 
-class ParsedChange {
-    type: string
+class ParsedLine {
+    // Line content
     content: string
+    // Trimmed line content
+    parsedContent: string
+    //  Line number
     ln: number
+    // Start/end of content within full content string
     range: Range = { start: 0, end: 0 }
 
-    constructor(change: parse.Change) {
-        this.type = change.type
-        this.content = change.content
-        this.ln = change.type === 'normal' ? change.ln2 : change.ln
-    }
-
-    parseContent() {
-        return this.type === 'normal'
-            ? this.content.trim()
-            : this.content.slice(1).trim()
+    constructor(line: usage.LineItem | parse.Change) {
+        this.content = line.content
+        this.parsedContent = this.content.trim()
+        this.ln = (line as usage.LineItem).ln
```

Comment:
whats going on here?

Reply:
The `ParsedLine` class is used as a common structure to work with both diff/file lines. The only problem is when extending this in `ParsedChangeLine`, `super()` is unhappy because normal diff lines don't have a `ln` property